db/db_flush_test.cc

```diff
@@ -78,6 +80,62 @@ TEST_F(DBFlushTest, FlushWhileWritingManifest) {
 #endif  // ROCKSDB_LITE
 }
 
+// We had issue when a flush with wait triggered due to ExportColumnFamily can
+// finish before the result is committed to manifest. This is happening because
+// of a simultaneous flush on the column family. Run in loop to reproduce the
+// failure.
+TEST_F(DBFlushTest, FlushBeforeWritingManifestWithCheckpoint) {
+  Options options;
+  options.disable_auto_compactions = true;
+  options.max_background_flushes = 3;
+  options.max_write_buffer_number = 3;
+  env_->SetBackgroundThreads(3, Env::HIGH);
+  options.env = env_;
+  Reopen(options);
+
+  SyncPoint::GetInstance()->LoadDependency(
+      {{"DBImpl::FlushMemTable:AfterScheduleNonExportFlush",
+        "FlushJob::WriteLevel0Table:flush_started2"}});
+
+  SyncPoint::GetInstance()->SetCallBack(
+      "FlushJob::WriteLevel0Table:flush_started1", [&](void* /*arg*/) {
+        // Add second write and flush without wait.
+        FlushOptions no_wait;
+        no_wait.wait = false;
+        no_wait.allow_write_stall = true;
+        ASSERT_OK(Put("foo", "v"));
+        ASSERT_OK(dbfull()->Flush(no_wait));
+      });
```

Comment:
My expectation is: with sync point, we don't need random sleep to reproduce a race condition in a **single fresh run** of a unit test. Can you try remove the random sleep in flush_job.cc?

Reply:
Even with interleaving it is not always guaranteed to repro this and hence I used random delay. I have removed random delay from flush_job.cc and have added it to a TEST_SYNC_POINT

Reply:
> Even with interleaving it is not always guaranteed to repro this and hence I used random delay

This totally defeats the purpose of syncpoint.